node/baremetal/README.md

```diff
@@ -0,0 +1,418 @@
+- OP-Geth - L2 execution client for Base
+- OP-Node - L2 rollup client for Base
+
+Base is built on top of the [Optimism Bedrock stack](https://community.optimism.io/docs/developers/bedrock/)
+which is why the clients are referred to as OP-Geth and OP-Node. They are run
+with ChainID 8453, which is the L2 chain ID for Base.
+
+## Hardware Requirements
+
+For running all 4 binaries (Eth Execution layer, Eth Consensus Layer, OP-Geth,
+and OP-Node) on mainnet, we recommend the following specs:
+
+- At least 32GB of RAM
+- At least 2 TB of storage
+- At least 4 CPU cores
+
+Some recommended sources for good devices for home-labbing include
+[Intel Nuc](https://www.intel.com/content/www/us/en/products/details/nuc.html)
+and [System76](https://system76.com/).
+
+## Software Requirements
+
+This guide was built using Ubuntu 22.04.3 LTS. All commands are going to be run
+using APT, but you can use your preferred package manager if on a different type of OS.
+
+This configuration utilizes systemd to run the nodes as services. This will
+ensure the nodes are always running in the background, and will automatically
+restart on a system reboot. It also provides journald for automatic logging.
+
+## Setup Ethereum
+
+In order to get the latest L2 chain data on your Base node, you will need to have
+an L1 source. This first section of the guide details setting up a full Ethereum
+node locally, but if you want to use a third party provider, there are several
+node providers listed [in the Base Documentation](https://docs.base.org/tools/node-providers).
+
+1. First, ensure your system is updated and upgraded:
+
+```bash
+sudo apt update && sudo apt upgrade -y
+```
+
+2. Install Geth:
+
+```bash
+sudo add-apt-repository -y ppa:ethereum/ethereum
+sudo apt install ethereum
+```
+
+3. Create a directory to hold the data for Ethereum:
+
+```bash
+sudo mkdir -p /opt/ethereum/geth/data
+sudo mkdir /opt/ethereum/prysm
+```
+
+4. Install Prysm
+
+```bash
+# if not already installed
+sudo apt install curl
+
+cd /opt/ethereum/prysm
+sudo curl https://raw.githubusercontent.com/prysmaticlabs/prysm/master/prysm.sh --output prysm.sh && chmod +x prysm.sh
+```
+
+and add a folder for data:
+
+```bash
+sudo mkdir /opt/ethereum/prysm/data
+```
+
+5. Create a JWT secret for Geth
+
+```bash
+cd /opt/ethereum/prysm
+sudo ./prysm.sh beacon-chain generate-auth-secret
+```
+
+6. Configure systemd to run Geth
+
+Add a new file called `/etc/systemd/system/geth.service`:
+
+```bash
+[Unit]
+Description=Geth Mainnet Ethereum Client
+Wants=prysm.service
+
+[Service]
+ExecStart=/usr/bin/geth --syncmode "snap" \
+        --http \
+        --http.api "eth,net,web3,personal" \
+        --cache=2048 \
+        --maxpeers=25 \
+        --datadir PATHTO/ethereum/geth/data \
+        --port 30303 \
+        --http.port 8545 \
+        --authrpc.jwtsecret /opt/ethereum/prysm/jwt.hex
+Restart=always
+RestartSec=60
+TimeoutSec=180
+KillMode=process
+
+[Install]
+WantedBy=multi-user.target
+```
+
```

Comment:
Shouldn't this have a chmod too?

Reply:
I didn't need it as they're installed with `sudo` and stored in a root directory `/opt`. Root user had permissions by default